extensions/indexer/indexer.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+type AcceptedIndexer interface {
+	Accepted(ctx context.Context, blk *chain.StatelessBlock) error
+}
+
+type SuccessfulTxIndexer interface {
+	Accepted(ctx context.Context, tx *chain.Transaction, result *chain.Result) error
+}
+
+type TxIndexer struct {
```

Comment:
If we're going to make this an extension that other vms will pull in it probably makes sense to add unit tests for this

Reply:
Wasn't sure if that should be part of this PR, but can add it

Reply:
I think we should make adding a unit test a separate PR because you currently need a builder in order to construct a stateless block...

We need to clean a lot of this code up to make writing unit tests more reasonable, so I'd prefer to defer this to a separate PR and keep the scope of this one a little bit smaller.